api/v1beta1/keystoneapi_types.go

```diff
@@ -269,10 +269,10 @@ func (instance KeystoneAPI) GetEndpoint(endpointType endpoint.Endpoint) (string,
 	return "", fmt.Errorf("%s endpoint not found", string(endpointType))
 }
 
-// IsReady - returns true if service is ready to server requests
+// IsReady - returns true if KeystoneAPI is reconciled successfully
 func (instance KeystoneAPI) IsReady() bool {
-	return instance.Status.Conditions.IsTrue(condition.ExposeServiceReadyCondition) &&
-		instance.Status.Conditions.IsTrue(condition.DeploymentReadyCondition)
+	readyCond := instance.Status.Conditions.Get(condition.ReadyCondition)
+	return readyCond != nil && readyCond.Status == corev1.ConditionTrue
```

Comment:
Could we just use `instance.Status.Conditions.IsTrue(condition.ReadyCondition)` here?  It seems like handles the `nil` case and then the boolean comparison [1].

[1] https://github.com/openstack-k8s-operators/lib-common/blob/d41eddbb4da64c307f3b510358f09a54a909b84d/modules/common/condition/funcs.go#L135-L142

Reply:
That is definitely better. Thanks for pointing that !